packages/react/src/hooks/accounts/useSigner.ts

```diff
@@ -7,19 +7,25 @@ import {
 } from '@wagmi/core'
 import * as React from 'react'
 
-import { QueryConfig } from '../../types'
+import { QueryConfig, QueryFunctionArgs } from '../../types'
 import { useQuery } from '../utils'
 
 export type UseSignerConfig = Omit<
   QueryConfig<FetchSignerResult, Error>,
   'cacheTime' | 'staleTime' | 'enabled'
->
+> & {
+  chainId?: number
+}
 
-export const queryKey = () => [{ entity: 'signer' }] as const
+export const queryKey = ({ chainId }: Partial<UseSignerConfig>) =>
+  [{ entity: 'signer', chainId, persist: false }] as const
 
-const queryFn = <TSigner extends Signer>() => fetchSigner<TSigner>()
+const queryFn = <TSigner extends Signer>({
+  queryKey: [{ chainId }],
+}: QueryFunctionArgs<typeof queryKey>) => fetchSigner<TSigner>({ chainId })
 
 export function useSigner<TSigner extends Signer>({
+  chainId,
```

Comment:
Should we use `const chainId = useChainId({ chainId: chainId_ })` instead of passing this `chainId` directly through?

Reply:
Yep, sounds good!